src/openforms/submissions/views.py

```diff
@@ -235,3 +240,41 @@ def get_sendfile_opts(self) -> dict:
             "mimetype": submission_attachment.content_type,
         }
         return opts
+
+
+class SearchSubmissionForCosignFormView(PermissionRequiredMixin, FormView):
+    form_class = SearchSubmissionForCosignForm
+    template_name = "submissions/find_submission_for_cosign.html"
+    raise_exception = True
+
+    cosign_url = None
+
+    def has_permission(self):
+        """
+        The user should have authenticated with the auth plugin specified on the form for the co-sign component
+        """
+        form = get_object_or_404(Form, slug=self.kwargs["form_slug"])
+        cosign_component = form.get_cosign_component()
+        expected_auth_plugin = cosign_component["authPlugin"]
+        return is_authenticated_with_plugin(self.request, expected_auth_plugin)
+
+    def get_initial(self):
+        return {"form_slug": self.kwargs["form_slug"]}
+
+    def form_valid(self, form):
+        submission = Submission.objects.get(
+            public_registration_reference=form.cleaned_data["code"]
+        )
```

Comment:
You can avoid this additional query because you're already doing it in the form clean method - so look up this submission (with all the relevant validation) and if all is find, you can store it in `self.cleaned_data["submission"]`

Then here in the view, it's just a `submission = form.cleaned_data["submission"]` access and continue business as usual.